src/snowflake/snowpark/mock/_plan.py

```diff
@@ -384,18 +384,33 @@ def handle_function_expression(
         type_hint = str(type_hints[key])
         keep_literal = "Column" not in type_hint
         if key == spec.varargs:
-            to_pass_args.extend(
-                [
-                    calculate_expression(
-                        c,
-                        input_data,
-                        analyzer,
-                        expr_to_alias,
-                        keep_literal=keep_literal,
+            # TODO is this really the best place for this?
```

Comment:
I feel this can be added to into `calculate_expression` as another top level if instance check

Reply:
I agree with Adam that this should be in `calculate_expression` instead. `calculate_expression` should be able to return a `TableEmulator` for expressions like `Star` and `SnowflakeUDTF`.

Reply:
Let's isolate this change into a separate PR so we can merge the rest of the fixes.